doc/developers/contributing.rst

```diff
@@ -619,16 +619,30 @@ Finally, follow the formatting rules below to make it consistently good:
         of the mentioned shapes. The default value is
         `np.ones(shape=(n_samples,))`.
 
+    list_param : list of int
+
+    typed_ndarray : ndarray of shape (n_samples,), dtype=np.int32
```

Comment:
when did we decide to put the dtype at the end, instead of `np.int32 ndarray of shape (n_samples,)`? This also doesn't play well with the other possible input types, if present.

Reply:
yeah  I prefer option 1

Reply:
I'm +1 for option 2. It's really not clear to me that `int list` is a list of int as it's contrary to typing conventions I am aware of , I would probably read that as `int` or `list`.

Reply:
I think I am starting prefer option 2. For the complicated case of `mixed_tuple_of_list`:

```py
mixed_tuple_of_list: list of (int,) or list of (int, int)
```

For array-like:


```py
d : array-like of int
```

For the various combinations of ndarrays:

```py
b : ndarray of shape (n_features,)

c : ndarray of int
```

I am on the fence about the following:

```py
a : ndarray of shape (n_features,) dtype int
```

I am hoping this is read as "ndarray of shape (n_features,) and of dtype int."